src/Service/Fleets.php

```diff
@@ -0,0 +1,267 @@
+     *
+     * @return int
+     *  A constant representing the result of the operation
+     */
+    public function addFleet($fleetName) {
+
+        $this->getFleetConfiguration();
+
+        if (array_key_exists($fleetName, $this->fleetConfig)) {
+            return self::FLEET_ALREADY_EXISTS;
+        }
+
+        $this->fleetConfig[$fleetName] = $this->defaultFleet();
+
+        $this->saveFleetsConfiguration();
+
+        return self::FLEET_ADDED;
+    }
+
+    /**
+     * Remove a fleet from this project
+     *
+     * @param string $fleetName
+     *  Name of the fleet to remove
+     *
+     * @return int
+     *  A constant representing the result
+     */
+    public function removeFleet($fleetName) {
+
+        $this->getFleetConfiguration();
+
+        if (!array_key_exists($fleetName, $this->fleetConfig)) {
+            return self::FLEET_DOES_NOT_EXIST;
+        }
+
+        unset($this->fleetConfig[$fleetName]);
+
+        $this->saveFleetsConfiguration();
+
+        return self::FLEET_REMOVED;
+    }
+
+    /**
+     * Add a project to a fleet
+     *
+     * @param string $fleetName
+     *  The name of the fleet to use
+     * @param string $projectID
+     *  The project ID
+     *
+     * @return int
+     *  A constant representing the result of the operation
+     */
+    public function addProject($fleetName, $projectID) {
+
+        $fleet = $this->getFleet($fleetName);
+
+        if ($fleet === self::FLEET_DOES_NOT_EXIST) {
+            return self::FLEET_DOES_NOT_EXIST;
+        }
+
+        if (in_array($projectID, $fleet)) {
+            return self::PROJECT_ALREADY_EXISTS;
+        }
+
+        $this->fleetConfig[$fleetName]['projects'][] = $projectID;
+
+        $this->saveFleetsConfiguration();
+
+        return self::PROJECT_ADDED;
+    }
+
+    /**
+     * Remove a project from a fleet
+     *
+     * @param string $fleetName
+     *  The name of the fleet to use
+     * @param string $projectID
+     *  The project ID
+     *
+     * @return int
+     *  A constant representing the result of the operation
+     */
+    public function removeProject($fleetName, $projectID) {
+
+        $fleet = $this->getFleet($fleetName);
+
+        if ($fleet === self::FLEET_DOES_NOT_EXIST) {
+            return self::FLEET_DOES_NOT_EXIST;
+        }
+
+        if (in_array($projectID, $fleet['projects'])) {
+
+            if (($key = array_search($projectID, $this->fleetConfig[$fleetName]['projects'])) !== false) {
+                unset($this->fleetConfig[$fleetName]['projects'][$key]);
+            }
+
+            $this->saveFleetsConfiguration();
+
+            return self::PROJECT_REMOVED;
+        }
+
+        return self::PROJECT_DOES_NOT_EXIST;
+    }
+
+    /**
+     * Get the name of the configuration file for fleets.
+     *
+     * @return string
+     */
+    protected function getConfigFileName()
+    {
+        return $this->config->get('service.project_config_dir') . '/fleets.yaml';
```

Comment:
This will load from `.platform/fleets.yaml`, relative to whatever directory the CLI is running in. Problems:

* The `local.fleet_config` option added in `config.yaml` in this branch isn't being used.
* The `local.fleet_config` (being defined as .platform/local/fleets.yaml` would be excluded by Git by default, that's probably not wanted.
* This should probably be based on the project root directory, which could be passed in by the caller, or obtained through the `LocalProject` service's `getProjectRoot()` method.

Reply:
I think local.fleet_config is a remnant of Christian's work. I'll remove that.

It's entirely possible that you might want to run fleet commands without a project. For that reason I remove any LocalProject references.

Reply:
OK, but selecting `.platform/fleets.yaml` from anywhere you run the command is going to be problematic, the base path (the cwd) could be anything and would often be invisible to the user depending on their terminal/prompt.

The root path should be something well defined: either the local project root, and/or somewhere in the user's home directory, and/or we allow the user to override the path to the fleets.yaml.